lib/proxy/router.go

```diff
@@ -381,8 +432,16 @@ func getServer(ctx context.Context, host, port string, site site) (types.Server,
 	}
 
 	strategy := types.RoutingStrategy_UNAMBIGUOUS_MATCH
+	matchPrincipal := func(principal string) bool {
+		return host == principal
+	}
 	if cfg, err := site.GetClusterNetworkingConfig(ctx); err == nil {
 		strategy = cfg.GetRoutingStrategy()
+		if cfg.GetCaseInsensitiveRouting() && isASCII(host) {
```

Comment:
Is there a reason this setting only applies in cases where the `host` is pure ASCII?